### PR TITLE
test(javascript/es6): Add test for es6

### PR DESCRIPTION
**Pull request by davixyz**
*Issue #, if available:*
https://github.com/amzn/style-dictionary/issues/140

*Description of changes:*
Add babel so jest can understand es6
https://jestjs.io/docs/en/getting-started.html#using-babel

Add test for es6

Note, this will probably keep failing until https://github.com/amzn/style-dictionary/pull/148 is merged

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: __tests__/formats/all.test.js [hunk 1]
```diff
@@ -39,11 +39,7 @@ describe('formats', () => {
   const globalDate = global.Date;
 
   beforeAll(() => {
-    global.Date = class extends Date {
```
Comment (davixyz) [inline __tests__/formats/all.test.js]: @dbanksdesign @chazzmoney @didoo I found a better way to do the constant date trick, in the previous way it was messing up when I added the ES6 support for JEST where it was kinda transforming this code and giving me a very cryptic error; But now it's fixed

**Comment by davixyz** [comment]
Adding ES6 support like this is trivial with babel-preset-env but if we want to go full on with Babel 7.0 we need to add way more dependencies like it explains here:
https://jestjs.io/docs/en/getting-started

Do we want to explore more in babel 7.0 or less dependencies?

**Comment by dbanksdesign** [comment]
TBH my babel/jest knowledge is pretty limited, what benefit would we get with using full on babel? I think I'm good with this PR and we can maybe do full on babel in the future if we need to. Thanks for doing this!

File: __tests__/formats/all.test.js (hunk 1 above)
Comment (dbanksdesign) [reply __tests__/formats/all.test.js]: As long as it works as expected! Nice work!

**Comment by davixyz** [comment]
@dbanksdesign If we use full babel for the whole library then we could essentially re-write it completely in ES6 and transpile it to ES5 on build. That's of course a big effort (similar to this jest effort). Now this addition of babel-preset-env is actually something we would need to do anyways when re-writing the whole library to ES6 so it's good, in this case Jest is smart enough to kinda understand the es6 and use the preset-env to parse the ES6 on runtime

**Comment by chazzmoney** [comment]
LGTM!

**Pull request merged**